File: NcrNpr.java
import java.util.Scanner;

public class NcrNpr {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in); 
  int n=sc.nextInt(); 
  int r=sc.nextInt(); 
   
  int result=fact(n)/fact(n-r)*fact(r); 
  System.out.println( "ncr "+result); 
   result=fact(n)/fact(n-r); 
  System.out.println("npr "+ result); 
 
 } 
 public static int fact(int n) { 
  int f=1; 
  for(int i=1;i<=n;i++) { 
   f=f*i; 
  } 
  return f; 
    }
}

File: ncr_npr.py
def factorial(n): 
    if n == 0: 
        return 1 
    return n * factorial(n - 1) 
 
n = int(input("Enter n: ")) 
r = int(input("Enter r: ")) 
 
ncr = factorial(n) // (factorial(r) * factorial(n - r)) 
npr = factorial(n) // factorial(n - r) 
 
print("NCR:", ncr) 
print("NPR:", npr) 